up-l1/mqtt_5.adoc

```diff
@@ -24,165 +24,215 @@ MQTT is an OASIS standard messaging protocol for the Internet of Things (IoT). I
+--
+* The MQTT payload **MUST** be the `UMessage.payload` field, which is a byte array to reduce size.
+--
 
-`{client_identifier}/{source.authority_name}/{source.ue_id}/{source.ue_version_major}/{source.resource_id}/{sink.authority_name}/{sink.ue_id}/{sink.ue_version_major}/{sink.resource_id}`
 
-If the no sink uuri is provided, then the topic sections using sink uuri segments are removed, like so:
+== MQTT5 Topics
 
-`{client_identifier}/{source.authority_name}/{source.ue_id}/{source.ue_version_major}/{source.resource_id}`
+The MQTT topic a message is published on utilizes the source and sink UUri fields. The topic is dependent on the use case for the transport implementation that will be discussed below.
 
-The client_identifier is used to determine where messages are coming from. The valid values for the client_identifier are:
 
-- `c` : Indicates that the message originated from the cloud
-- `d` : Indicates that the message originated from a device and is remote
+=== uE-2-uE Communication
 
-The client_identifier that is used for the topic is based on the kind of MQTT uTransport client that is sending the message. An MQTT uTransport client can be one of the following:
+When a message is sent from one uE to another uE, the topic is constructed using the source and sink UUri fields. The topic is constructed as follows using the string representation of a `UUri`:
 
-- Cloud MQTT uTransport client: A client that runs in the cloud.
-- Remote MQTT uTransport client: A client that runs on a device and is connected to a cloud MQTT broker.
+[.specitem,oft-sid="req~up-transport-mqtt5-ue2ue-topic~1",oft-needs="impl,utest"]
+--
+`{UAttributes.source}/{UAttributes.sink}/`
+--
 
-Below are examples of messages being sent and how to subscribe to those messages:
+[.specitem,oft-sid="req~up-transport-mqtt5-ue2ue-topic-nosink~1",oft-needs="impl,utest"]
+--
+If the messages does not have a sink `UUri`, then the sink portion of the MQTT5 topic *MUST* be omitted.
+--
 
-==== Remote uTransport Client Examples
+==== Examples
 
-How a source and sink UUri are mapped to an MQTT topic when the message is sent from a remote uTransport client:
-
-[cols="1,1,1"]
+.uE-2-uE Communication Topics
+[cols="1,2,2,4"]
 |===
-| source | sink | topic
+| Type| source URI | sink URI | MQTT5 Topic
 
-| //vehicle_1.veh.com/AB34/1/12CD | //vehicle_2.veh.com/43BA/1/DC21 | {cli_identifier}/vehicle_1.veh.com/AB34/1/12CD/vehicle_2.veh.com/43BA/1/DC21
-| //vehicle_1.veh.com/AB34/1/12CD | None | {cli_identifier}/vehicle_1.veh.com/AB34/1/12CD
+| *Request* | `//device1/AB34/1/0` | `//device1/43BA/1/2` | `device1/AB34/1/0/device1/43BA/1/2`
+| *Response* | `//device1/43BA/1/2` | `//device1/AB34/1/0` | `device1/43BA/1/2/device1/AB34/1/0`
+| *Publish* | `//device1/AB34/1/8000` | None | `device1/AB34/1/8000`
+| *Notification* | `//device1/43BA/1/8001` | `//device1/AB34/1/0` | `device1/43BA/1/8001/device1/AB34/1/0`
```

Comment:
these examples do not match the topic pattern defined above

Reply:
with a source of `//device1/43BA/1/8001` and a sink of `//device1/AB34/1/0` and a pattern of `$(source}/${sink}`, the topic should be `//device1/43BA/1/8001///device1/AB34/1/0`, shouldn't it? However, in the table, you are using `device1/43BA/1/8001/device1/AB34/1/0`.

Even considering that the string representation of a UUri without an authority name actually is not `//device1/43BA/1/8001` but `/device1/43BA/1/8001`, applying the pattern would yield `/device1/AB34/1/8000//device1/AB34/1/0` ...

Reply:
OK I see your point. We cannot simply append the strings so I will go back to listing each attribute independently.

Reply:
done